### PR TITLE
chore: Adds logs messages for detailed debugging.

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
This PR adds log messages to community templates publish flow to help erratic failures 
## Type of change
- Chore (housekeeping or task changes that don't impact user perception)
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java [hunk 1]
```diff
@@ -369,14 +371,21 @@ private Mono<Application> updateApplicationFlags(String applicationId, String br
     public Mono<Application> publishAsCommunityTemplate(CommunityTemplateDTO resource) {
         return exportApplicationService
                 .exportApplicationById(resource.getApplicationId(), resource.getBranchName())
-                .flatMap(appJson -> uploadCommunityTemplateToCS(
-                        createCommunityTemplateUploadDTO(resource.getApplicationId(), appJson, resource)))
+                .flatMap(appJson -> {
+                    Mono<ApplicationTemplate> template = uploadCommunityTemplateToCS(
+                            createCommunityTemplateUploadDTO(resource.getApplicationId(), appJson, resource));
+                    log.info("CommunityTemplatePublish [{}] 2. uploaded to CS", resource.getApplicationId());
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:377]: We need to handle the error from the CS API to get meaningful logs. Please add error handler in the the method `uploadCommunityTemplateToCS`

Refer this - https://github.com/appsmithorg/appsmith/blob/8b5269fef1f7a108e436bbf4ecf0d13e345877dc/app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java#L354

**Review by AnaghHegde — CHANGES_REQUESTED** [review]
Nit - Can you please update the PR with meaningful information? And you can remove any section if it does not make sense. Thanks :)

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java [hunk 2]
```diff
@@ -351,7 +354,8 @@ private Mono<ApplicationTemplate> uploadCommunityTemplateToCS(CommunityTemplateU
                 .accept(MediaType.APPLICATION_JSON)
                 .body(BodyInserters.fromValue(payload))
                 .retrieve()
-                .bodyToMono(ApplicationTemplate.class);
+                .bodyToMono(ApplicationTemplate.class)
+                .onErrorResume(WebClientResponseException.class, ex -> Mono.error(ex));
```
Comment (rahulbarwal) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:358]: @AnaghHegde would this be sufficient to handle errors?

**Comment by rahulbarwal** [comment]
This PR is not required anymore. This https://github.com/appsmithorg/appsmith/pull/28347 handles it.

**Pull request closed without merging**